indico/modules/categories/controllers/display.py

```diff
@@ -348,3 +352,171 @@ def _process(self):
                                       session.user,
                                       Event.end_dt >= now_utc())
         return send_file(filename, buf, 'application/atom+xml')
+
+
+class RHCategoryOverview(RHDisplayCategoryBase):
+    """Display the events for a particular day, week or month"""
+
+    def _checkParams(self):
+        RHDisplayCategoryBase._checkParams(self)
+        self.detail = request.args.get('detail', 'event')
+        if self.detail not in ('event', 'session', 'contribution'):
+            raise BadRequest('Invalid detail argument')
+        self.period = request.args.get('period', 'day')
+        if self.period not in ('day', 'month', 'week'):
+            raise BadRequest('Invalid period argument')
+        if 'date' in request.args:
+            try:
+                date = datetime.strptime(request.args['date'], '%Y-%m-%d')
+            except ValueError:
+                raise BadRequest('Invalid date argument')
+        else:
+            date = datetime.now()
+        date = self.category.display_tzinfo.localize(date)
+        date = date.replace(hour=0, minute=0, second=0, microsecond=0)
+        if self.period == 'day':
+            self.start_dt = date
+            self.end_dt = self.start_dt + relativedelta(days=1)
+        elif self.period == 'week':
+            self.start_dt = date - relativedelta(days=date.weekday())
+            self.end_dt = self.start_dt + relativedelta(days=7)
+        elif self.period == 'month':
+            self.start_dt = date + relativedelta(day=1)
+            self.end_dt = self.start_dt + relativedelta(months=1)
+
+    def _process(self):
+        info = get_category_timetable([self.category.id], self.start_dt, self.end_dt, detail_level=self.detail,
+                                      tz=self.category.display_tzinfo, from_categ=self.category, grouped=False)
+        events = info['events']
+
+        # Only categories with icons are listed in the sidebar
+        subcategories = {event.category for event in events if event.category.has_icon}
+
+        # Python doesn't have flat_map() and it improves the readability of the line bellow greatly
```

Comment:
*below, and i'd move it to a proper method or a module-level function. and `imap` instead of `map` would probably be nicer here